api/handlers.go

```diff
@@ -5,42 +5,193 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
-	"os"
-	"path/filepath"
+	"strings"
 
-	"github.com/qri-io/dataset"
 	"github.com/qri-io/qri/api/util"
 	"github.com/qri-io/qri/base/archive"
+	"github.com/qri-io/qri/dsref"
 	"github.com/qri-io/qri/lib"
-	"github.com/qri-io/qri/profile"
-	reporef "github.com/qri-io/qri/repo/ref"
 )
 
 // GetHandler is a dataset single endpoint
 func GetHandler(inst *lib.Instance, routePrefix string) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		if !(r.Method == http.MethodGet || r.Method == http.MethodPost) {
+		if r.Method != http.MethodGet {
 			util.NotFoundHandler(w, r)
 			return
 		}
-		params := &lib.GetParams{}
 
-		err := UnmarshalParams(r, params)
+		var err error
+		returnCase, params, err := getCaseAndParamsFromRequest(r)
 		if err != nil {
 			util.WriteErrResponse(w, http.StatusBadRequest, err)
 			return
 		}
-		got, _, err := inst.Dispatch(r.Context(), "dataset.get", params)
+
+		switch returnCase {
+
+		case "zip":
+			outBytes, err := inst.Dataset().GetZip(r.Context(), params)
+			if err != nil {
+				util.RespondWithError(w, err)
+				return
+			}
+			writeFileResponse(w, outBytes, "dataset.zip", "zip")
```

Comment:
It was kinda nice before how downloading a zip of a dataset would have the dataset name (and commit timestamp, I believe?) in the filename that got downloaded. An unfortunate consequence of simplifying the GetResult is that we lose the ability to do that. I don't know if it's worth the effort to add that back in, just wanted to add it as a data point.

Reply:
Ahh yes, this was another tacit change that I never surfaced to the group, thank you. 

One solution I thought of was `lib.GetZip` could return `ZipResults`
```
type ZipResults struct {
  Bytes []bytes
  Name string // generated name
}
```